console/v1/types_console_link.go

```diff
@@ -19,6 +19,8 @@ type ConsoleLinkSpec struct {
 	Link `json:",inline"`
 	// location determines which location in the console the link will be appended to.
 	Location ConsoleLinkLocation `json:"location"`
+	// section is the section of the ConsoleLinkLocation in which the link should appear.
```

Comment:
We need to make it clear that this is only used for `ApplicationMenu` links.

We should enumerate the possible values as this is well defined.

Reply:
Maybe `applicationMenuSection` to make it really clear it's only for application menu links.

Reply:
Are we sure that the `HelpMenu` and `UserMenu` wont have sections in the future ?

Reply:
It seems unlikely to me... It's possible we add a new type in the future that has sections, though.

The problem if we leave it general is it's hard to add validation. Ideally we'd validate that the application menu section is one of the 3 values we allow, and the sections will likely be different for different link types.

Reply:
yeah makes sense... then the `applicationMenuSection` would make sense... it could even be an enum with:
- `Red Hat Application`
- `3rd Party Application`
- `Customer Application`

Reply:
Also shouldn't `Link` type be extended with the `Icon` ? since the links in the ApplicationMenu will contain icons ?

Reply:
I'd add icon in a follow-on since that makes things a lot more complicated. You have to deal with referencing images in config maps potentially, unless we restrict it to URLs and data URIs.

Reply:
Hm, maybe it should be

```yaml
spec:
  location: 'ApplicationMenu'
  href: 'https://www.example.com'
  text: My Application
  applicationMenu:
    section: RedHat
    imageURL: 'https://www.example.com/my-icon.svg'
```

Reply:
I spoke briefly with @serenamarie125 about this. I don't think we want an enum. There could be desire to have additional sections or have the name of the section changed (especially if a new menu type is added in the future).

Reply:
One other question here...

Assuming we always want the ordering above, how will the UI know how to order the sections?

Reply:
> One other question here...
> 
> Assuming we always want the ordering above, how will the UI know how to order the sections?

Maybe special case the special sections and then alphabetize the rest?